### DISCUSSION TITLE
Get SharePoint online user access token without registering as app

### DISCUSSION THREAD

**Discussion by krunal039** [pnp/pnpcore#390] [Q&A] [answered]
I am looking at way to get user token with userid and password for sharepoint online via c#. I can register as app with clientid and client secret but I am not looking for it.

PnP has powershell commend Get-PnPAccessToken and just need same via C#.

Is there a way to get it for SharePoint online via c#?

Many thanks,

**Comment by jansenbe** [comment, marked as answer]
To get an access token you need an app, there's no way around. In PnP PowerShell you might not have created an app but in that case you're reusing an other app like the PnP Management Shell app or the SPO Management Shell app. You'll then get an access token with the permissions defined on the app. I would recommend creating the needed app in Azure AD with the permissions you need and then use that. Register-PnPAzureADApp can help you with setting up that app

**Reply by krunal039** [reply]
@jansenbe Thanks for it. I manage to get it working as you suggested avove.
